File: data/data_transpiled/at_coder/abc110/C/4929388.java
public static void input ( Scanner in ) {
  String s = in . nextLine ( ) ;
  String t = in . nextLine ( ) ;
  HashMap < String , String > dic1 = new HashMap < String , String > ( ) ;
  HashMap < String , String > dic2 = new HashMap < String , String > ( ) ;
  boolean f = true ;
  for ( int i = 0 ;
  i < s . length ( ) ;
  i ++ ) {
    String j = in . nextLine ( ) ;
    if ( dic1 . containsKey ( j ) ) {
      if ( dic1 . get ( j ) != i ) {
        f = false ;
        break ;
      }
    }
    dic1 . put ( j , s ) ;
    if ( dic2 . containsKey ( j ) ) {
      if ( dic2 . get ( j ) != i ) {
        f = false ;
        break ;
      }
    }
    dic2 . put ( j , t ) ;
  }
  if ( f ) {
    in . close ( ) ;
    System . out . println ( "Yes" ) ;
  }
  else {
    in . close ( ) ;
  }
}

File: storage/data_original/at_coder/abc110/C/4929388.py
s = input()
t = input()
dic1 = {}
dic2 = {}
f = True
for i, j in zip(s, t):
    if i in dic1:
        if dic1[i] != j:
            f = False
            break
    dic1[i] = j
    if j in dic2:
        if dic2[j] != i:
            f = False
            break
    dic2[j] = i
if f:
    print('Yes')
else:
    print('No')